packages/babel-traverse/src/path/lib/removal-hooks.js

```diff
@@ -64,5 +64,18 @@ export let hooks = [
       }
       return true;
     }
+  },
+
+  function (self, parent) {
+    if (
+      (self.key === 'consequent' || self.key === 'alternate') && parent.isIfStatement() ||
+      self.key === 'body' && (parent.isForStatement() || parent.isWhileStatement() || parent.isDoWhileStatement())
```

Comment:
This will skip `for..in`, `for..of`, I'd recommend doing

    parent.isLoop()

e.g.

```
if ((parent.isIfStatement() && (self.key === 'consequent' || self.key === 'alternate')) ||
  (parent.isLoop() && self.key === 'body')){